tools/d2g/d2g.go

```diff
@@ -57,22 +57,25 @@ func ConvertTaskDefinition(dwTaskDef json.RawMessage) (json.RawMessage, error) {
 		return nil, fmt.Errorf("cannot convert Docker Worker payload: %v", err)
 	}
 
-	if scopes, exists := parsedTaskDef["scopes"]; exists {
+	if scopes, exists := parsedDwTaskDef["scopes"]; exists {
 		var dwScopes []string
 		for _, scope := range scopes.([]interface{}) {
 			dwScopes = append(dwScopes, scope.(string))
 		}
-		parsedTaskDef["scopes"] = Scopes(dwScopes)
+		parsedDwTaskDef["scopes"], err = Scopes(dwScopes, dwTaskDef)
+		if err != nil {
+			return nil, fmt.Errorf("cannot convert Docker Worker scopes: %v", err)
+		}
 	}
 
 	d2gConvertedPayloadJSON, err := json.Marshal(*gwPayload)
 	if err != nil {
 		return nil, fmt.Errorf("cannot marshal Generic Worker payload: %v", err)
 	}
 
-	parsedTaskDef["payload"] = json.RawMessage(d2gConvertedPayloadJSON)
+	parsedDwTaskDef["payload"] = json.RawMessage(d2gConvertedPayloadJSON)
 
-	gwTaskDef, err = json.MarshalIndent(parsedTaskDef, "", "  ")
+	gwTaskDef, err = json.MarshalIndent(parsedDwTaskDef, "", "  ")
```

Comment:
the left side says it is generic-worker task definition, but right side says it is a docker worker task definition?

Reply:
Thanks, just a typo, good catch!